src/stores/LifecycleStore.ts

```diff
@@ -51,6 +59,11 @@ class LifecycleStore extends AsyncStore<IState> {
                 });
                 break;
             case "MatrixActions.sync": {
+                if (payload.state === SyncState.Syncing && payload.prevState !== SyncState.Syncing) {
```

Comment:
If we're in the territory of quick fixes, it seems like the main thing that is actually going to make a difference here is removing the `await` from the call to `checkServerVersions`. Moving it to the `MatrixActions.sync` handler maybe makes sense generally, but was there a particular reason for that decision?

Reply:
So that we got some form of retry rather than just firing off the function once and forgetting about it. With the await, whilst it broke the app if we couldn't connect, this meant we had always checked the server versions if we logged in, whereas if I just removed the await then we could end up logged in but not having done the server versions check.

Reply:
right, ok.